src/main/java/com/uwaterloo/connect/model/User.java

```diff
@@ -0,0 +1,89 @@
+package com.uwaterloo.connect.model;
+
+import com.uwaterloo.connect.enums.UserRole;
+import jakarta.persistence.*;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Collection;
+import java.util.Collections;
+
+@Getter
+@Setter
+@EqualsAndHashCode
+@Entity
+@Table(name = "users")
+@NoArgsConstructor
+public class User implements UserDetails {
+
+    @Id
+    @SequenceGenerator(name = "user_sequence", sequenceName = "user_sequence", allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "user_sequence")
+    private Long id;
+    private String firstName;
+    private String lastName;
+    private String userName;
+    private String email;
+    private String password;
+    private LocalDate dateOfBirth;
+    private boolean active = false; // will enable once they confirm email
+    private boolean locked = false;
+    @Enumerated(EnumType.STRING)
+    private UserRole userRole;
+    private LocalDateTime createdAt = LocalDateTime.now();;
+    private LocalDateTime updatedAt = LocalDateTime.now();;
+
+    public User(String firstName, String lastName, String userName, String email, String password, LocalDate dateOfBirth, UserRole userRole) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.userName = userName;
+        this.email = email;
+        this.password = password;
+        this.dateOfBirth = dateOfBirth;
+        this.userRole = userRole;
+    }
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        SimpleGrantedAuthority authority = new SimpleGrantedAuthority(userRole.name());
+        return Collections.singletonList(authority);
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public String getUsername() {
+        return email;
+    } // this will be used for authentication purposes by spring security
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return !locked;
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
```

Comment:
@JobanThind , Is this supposed to be hard coded? I also cant find any usages

Reply:
This is done to implement UserDetails of Spring security, and we are not using this attribute as per our model and giving false will fail the authentication.